libzecale/circuits/pairing/bls12_377_pairing.hpp

```diff
@@ -441,6 +464,71 @@ class bls12_377_final_exp_last_part_gadget
     void generate_r1cs_witness();
 };
 
+// Wrapper around final_exp gadgets with interface expected by the groth16
+// gadgets. `result_is_one` is constrained to a boolean (0 or 1), and set in
+// 'generate_r1cs_witness' based on the output value of the final
+// exponentiation (if final exp == 1, `result_is_one` is set to 1, otherwise
+// `result_is_one` is set to 0).
+//
+// Note that the constraints on the final exp output are ONLY enforced when
+// `result_is_one` == 1. In otherwords, it is infeasible to generate valid
+// inputs such that the final exp output is not equal to 1 and result_is_one ==
+// 1. However, it IS possible to generate inputs such that final_exp == 1 but
```

Comment:
Shouldn't it be "However, it IS possible to generate inputs such that the final exp output is not equal to 1 but `result_is_one` == 0"?

Reply:
If `result_is_one == 0`, all checks are effectively disabled and so it IS possible that `final_exp == 1`.
I.e. (if I've understood) `result_is_one == 1` means that the final exp value is 1, but `result_is_one == 0` does not say anything about the value of final exp.

Reply:
As discussed, seems like we are saying the same thing. If `result_is_one = 0` then the following checks are rendered dummy, which means that the result isn't constrained to a given value.